(#2) Adds Initial Examples (#3)
chocolatey/ccm-deployment-examples: deployments/upgrading-chocolatey-for-business-clients/README.md

# Upgrading Chocolatey for Business Client Components

## Purpose

These Deployment Plans are examples of upgrading the Chocolatey for Business client components.

There are two example Deployment Plans here. They follow the instructions laid out on the [upgrading to Chocolatey v2](https://docs.chocolatey.org/en-us/guides/upgrading-to-chocolatey-v2-v6#upgrade-using-chocolatey-central-management-deployments) page.

The first upgrades the Chocolatey for Business client components to the latest 1.x version.

https://github.com/chocolatey/ccm-deployment-examples/blob/b79340df0363be0fcda3192828cef96095320d3d/deployments/upgrading-chocolatey-for-business-clients/DeploymentPlan-Latest1x.json#L1-L34

The second upgrades Chocolatey CLI to the latest version.

https://github.com/chocolatey/ccm-deployment-examples/blob/b79340df0363be0fcda3192828cef96095320d3d/deployments/upgrading-chocolatey-for-business-clients/DeploymentPlan-Latest2x.json#L1-L34

## Usage

> You should import both the `Latest1x` and `Latest2x` deployments if you are on a version of Chocolatey CLI older than 1.4.0, and deploy `Upgrade Chocolatey for Business Client to Latest 1.x` first, before deploying `Upgrade Chocolatey for Business Client` to upgrade to the latest version.

Reviewer: ```suggestion
You should import both the `Latest1x` and `Latest2x` Deployment Plans if you are on a version of Chocolatey CLI older than 1.4.0, and deploy `Upgrade Chocolatey for Business Client to Latest 1.x` first, before deploying `Upgrade Chocolatey for Business Client` to upgrade to the latest version.
```

# Upgrading Chocolatey for Business Client Components

## Purpose

> This package is an example of upgrading the Chocolatey for Business client components.

Reviewer 2: ```suggestion
These Deployment Plans are examples of upgrading the Chocolatey for Business client components.
```

There are two example deployments here. They follow the instructions laid out on the [upgrading to Chocolatey v2](https://docs.chocolatey.org/en-us/guides/upgrading-to-chocolatey-v2-v6#upgrade-using-chocolatey-central-management-deployments) docs page.

The first upgrades the Chocolatey for Business client components to the latest 1.* version.

https://github.com/chocolatey/ccm-deployment-examples/blob/b79340df0363be0fcda3192828cef96095320d3d/deployments/upgrading-chocolatey-for-business-clients/Deployment-Latest1x.json#L1-L34

The second upgrades Chocolatey to the latest version.

> https://github.com/chocolatey/ccm-deployment-examples/blob/b79340df0363be0fcda3192828cef96095320d3d/deployments/upgrading-chocolatey-for-business-clients/Deployment-Latest2x.json#L1-L34

Reviewer: ```suggestion
There are two example Deployment Plans here. They follow the instructions laid out on the [upgrading to Chocolatey v2](https://docs.chocolatey.org/en-us/guides/upgrading-to-chocolatey-v2-v6#upgrade-using-chocolatey-central-management-deployments) page.

The first upgrades the Chocolatey for Business client components to the latest 1.x version.

https://github.com/chocolatey/ccm-deployment-examples/blob/b79340df0363be0fcda3192828cef96095320d3d/deployments/upgrading-chocolatey-for-business-clients/Deployment-Latest1x.json#L1-L34

The second upgrades Chocolatey CLI to the latest version.

https://github.com/chocolatey/ccm-deployment-examples/blob/b79340df0363be0fcda3192828cef96095320d3d/deployments/upgrading-chocolatey-for-business-clients/Deployment-Latest2x.json#L1-L34
```

## Usage

> You should import both the `Latest1x` and `Latest2x` deployments if you are on a version of Chocolatey older than 1.4.0, and deploy `Upgrade Chocolatey for Business Client to Latest 1.*` first, before deploying `Upgrade Chocolatey for Business Client` to upgrade to the latest version.

Reviewer: ```suggestion
You should import both the `Latest1x` and `Latest2x` deployments if you are on a version of Chocolatey CLI older than 1.4.0, and deploy `Upgrade Chocolatey for Business Client to Latest 1.x` first, before deploying `Upgrade Chocolatey for Business Client` to upgrade to the latest version.
```

If you are upgraded to Chocolatey CLI version 2.0 or above, you can just import and run `Latest2x`.

> > Although the ScheduledTasks module is available on Windows Server 2012 R2, the chocolatey-agent service encounters an error when trying to import it. It is recommended to explore other options for the scheduled task if you're using Windows Server 2012 R2.

Reviewer: ```suggestion
> Although the ScheduledTasks module is available on Windows Server 2012 R2, the `chocolatey-agent` service encounters an error when trying to import it. It is recommended to explore other options for the scheduled task if you're using Windows Server 2012 R2.
```

Reviewer: What other options?

Author: Jokingly: Upgrading to a more recent OS.

In more seriousness: We don't currently have company recommendations for any alternatives to this. I thought it prudent to include the warning, as I found it interesting when reading the docs, but would rather remove the warning than add recommendations for an SKU that had (initial) extended support end today (🎉). If we really want, I can investigate and create a `DeploymentPlan-2012R2Only.json` with an alternative? I think I have a few options available...

> After importing the deployments, you will need to set both the target groups. We strongly recommend testing this on a smaller group to ensure it works nicely in your environment.
> 
> ### Importing the Deployment
> 
> - Download the `Deployment-Latest1x.json` and `Deployment-Latest2x.json` from this directory
> 
> - Login to your CCM instance
> 
> - Navigate to the `Deployments` tab (on the left navigation bar) and click `Import Deployment`  
>     ![The Deployments selection on the left nav bar, and the Import Deployment button](/images/DeploymentTab.jpg)
> 
> - Click `Browse` and select the downloaded json files  
>     ![The Import Deployment Modal](/images/DeploymentImportModal.jpg)
> 
> - Click `Import`
> 
> For further details, please see the [documentation at docs.chocolatey.org](https://docs.chocolatey.org/en-us/central-management/usage/examples/deployments).

Reviewer: ```suggestion
After importing the Deployment Plans, you will need to set the Target Groups for each. We strongly recommend testing this on a smaller group to ensure it works as expected in your environment.

### Importing the Deployment Plans

- Download the `Deployment-Latest1x.json` and `Deployment-Latest2x.json` from this directory.

- Login to your Chocolatey Central Management instance.

- Navigate to the `Deployments` tab (on the left navigation bar) and click `Import Deployment`  .
    ![The Deployments selection on the left nav bar, and the Import Deployment button](/images/DeploymentTab.jpg)

- Click `Browse` and select the downloaded `.json` files.
    ![The Import Deployment Modal](/images/DeploymentImportModal.jpg)

- Click `Import`.

For further details, please see the [documentation](https://docs.chocolatey.org/en-us/central-management/usage/examples/deployments).
```

Author: ```suggestion
After importing the Deployment Plans, you will need to set the Target Groups for each. We strongly recommend testing this on a smaller group to ensure it works as expected in your environment.

### Importing the Deployment Plans

- Download the `Deployment-Latest1x.json` and `Deployment-Latest2x.json` from this directory.

- Login to your Chocolatey Central Management instance.

- Navigate to the `Deployments` tab (on the left navigation bar) and click `Import Deployment`.  
    ![The Deployments selection on the left nav bar, and the Import Deployment button](/images/DeploymentTab.jpg)

- Click `Browse` and select the downloaded `.json` files.  
    ![The Import Deployment Modal](/images/DeploymentImportModal.jpg)

- Click `Import`.

For further details, please see the [documentation](https://docs.chocolatey.org/en-us/central-management/usage/examples/deployments).
```

### Setting the Target Group

You will need to set a target group for the deployment. 

To do so:

- Mouse-over the deployment step and click the `Edit` button:
    ![Editing a Deployment Step](/images/EditDeploymentStep.jpg)

- Navigate to the `Select Target Groups` panel

- Highlight your preferred groups and click the `>` button to move them into `Selected Groups`
    ![Selecting Target Groups](/images/SelectingTargetGroups.jpg)

> - Click `Save`

Reviewer: ```suggestion
You will need to set a Target Group for the Deployment Steps:

- Mouse-over the Deployment Step and click the `Edit` button.
    ![Editing a Deployment Step](/images/EditDeploymentStep.jpg)

- Navigate to the `Select Target Groups` panel.

- Highlight your preferred groups and click the `>` button to move them into `Selected Groups`.
    ![Selecting Target Groups](/images/SelectingTargetGroups.jpg)

- Click `Save`.
```
